crypto/encode_decode/encoder_lib.c

```diff
@@ -49,6 +49,43 @@ int OSSL_ENCODER_to_fp(OSSL_ENCODER_CTX *ctx, FILE *fp)
 }
 #endif
 
+int OSSL_ENCODER_to_data(OSSL_ENCODER_CTX *ctx, unsigned char **pdata,
+                         size_t *pdata_len)
+{
+    BIO *out = BIO_new(BIO_s_mem());
+    BUF_MEM *buf = NULL;
+    int ret = 0;
+
+    if (pdata_len == NULL) {
+        ERR_raise(ERR_LIB_OSSL_DECODER, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
+    }
+
+    if (OSSL_ENCODER_to_bio(ctx, out)
```

Comment:
Should there be explicit check for `out == NULL` rather than depending on OSSL_ENCODER_to_bio to fail?

Reply:
I don't see why...  this code isn't using `out` directly, and I think it better to leave it to those routines that *do* derrefence it directly to do the checks and return an appropriate error state.  If check for NULL defensively at every level, we risk covering up latent bugs further down.

Reply:
The problem is that in this case you'll get rather weird errors on top of the error stack rather than the interesting one that memory allocation failed, which will be somewhere below.

Reply:
Hmm?  When `out` is NULL?  Are you perhaps thinking of OSSL_DECODER, where the guessing structure does lead to that sort of error stack?  If not, do you have something to demonstrate this with?  I'd very much like to see what you see...

Reply:
I suppose it will make it go down to BIO_write() calls to NULL bio which actually does not put anything on error stack, so you're probably right that there won't be anything added onto the stack. OK, then. But yeah, the decoder case should be fixed.

Reply:
It could have pushed ERR_R_PASSED_NULL_ARGUMENT, or something less subtle like (non-existing for the moment) BIO_R_INVALID_BIO...  I don't see how that would have been a mystery...

(and I do see the lack of error reporting in BIO as a bug, mind you)

Reply:
I am not saying it would be a "mystery". Just that if this function directly returned when the BIO allocation fails the topmost error code would already provide the most accurate information on the failure and you would not have to look down into the error stack for it. But yeah, this is not a deal breaker (even for the decoder function where we have the same situation), so I am not going to fight for it.

Reply:
@t8m is right here - its a malloc failure at the top level